scheduler.go

```diff
@@ -340,6 +348,19 @@ func (s *scheduler) selectExecJobsOutCompleted(id uuid.UUID) {
 		return
 	}
 
+	// if the job has more than one nextScheduled time,
+	// we need to remove any that are in the past.
+	if len(j.nextScheduled) > 1 {
```

Comment:
Wouldn't you want to do this all the time? Even if `nextScheduled` had one element, and the timestamp on the one element was in the past, wouldn't you skip that execution?

Reply:
So, this doesn't impact scheduling here because it's always based on the the diff from the last nextScheduled value. I had a reason to keep it if there was only 1 value, which was so that calling Job.NextRun would always return something, but a next run in the past is pretty useless.